packages/components/src/dialog/index.tsx

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import type { ReactNode } from 'react';
+import Modal from 'react-modal';
+import classnames from 'classnames';
+import config from '@automattic/calypso-config';
+
+/**
+ * Internal dependencies
+ */
+import ButtonBar from './button-bar';
+import type { Button, BaseButton } from './button-bar';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+type Props = {
+	additionalClassNames: Parameters< typeof classnames >[ 0 ];
```

Comment:
Curious: Is this a common idiom for getting a type? This looks a bit confusing at first glance, but maybe I'm just not used to it. I see it evaluates to `ClassValue`.

Reply:
It shouldn't be common, but unfortunately some libraries don't export their types (`classnames` being one of them), so to get the type for `ClassValue` you have to use the `Parameters` utility type. A similar type, `ReturnType` also exists for doing the same thing but for return values.